tests/src/Kernel/DataProducer/EntityDefinitionTest.php

```diff
@@ -201,304 +551,77 @@ public function testEntityDefinition() {
-            'maxNumItems' => 1,
-            'status' => TRUE,
-            'defaultValue' => '',
-            'isReference' => FALSE,
-            'isHidden' => FALSE,
-            'weight' => 16,
-            'settings' => NULL,
-          ],
-          15 =>
-          [
-            'id' => 'default_langcode',
-            'label' => 'Default translation',
-            'description' => 'A flag indicating whether this is the default translation.',
-            'type' => 'boolean',
-            'required' => FALSE,
-            'multiple' => FALSE,
-            'maxNumItems' => 1,
-            'status' => TRUE,
-            'defaultValue' => '1',
-            'isReference' => FALSE,
-            'isHidden' => FALSE,
-            'weight' => 0,
-            'settings' => NULL,
-          ],
-          16 =>
-          [
-            'id' => 'revision_default',
-            'label' => 'Default revision',
-            'description' => 'A flag indicating whether this was a default revision when it was saved.',
-            'type' => 'boolean',
-            'required' => FALSE,
-            'multiple' => FALSE,
-            'maxNumItems' => 1,
-            'status' => TRUE,
-            'defaultValue' => '',
-            'isReference' => FALSE,
-            'isHidden' => FALSE,
-            'weight' => 0,
-            'settings' => NULL,
-          ],
-          17 =>
-          [
-            'id' => 'revision_translation_affected',
-            'label' => 'Revision translation affected',
-            'description' => 'Indicates if the last edit of a translation belongs to current revision.',
-            'type' => 'boolean',
-            'required' => FALSE,
-            'multiple' => FALSE,
-            'maxNumItems' => 1,
-            'status' => TRUE,
-            'defaultValue' => '',
-            'isReference' => FALSE,
-            'isHidden' => FALSE,
-            'weight' => 0,
-            'settings' => NULL,
-          ],
-        ],
-      ],
-    ]
-    );
+    $this->defaultCacheTags[] = 'config:field.field.node.article.field_test';
+    $this->assertResults($query, [], $this->fullDefinitionResult);
+  }
+
+  /**
+   * Tests that filtering base field definitions works.
+   */
+  public function testBaseFieldFilter() {
+    $query = <<<GQL
+      query {
+        entityDefinition(entity_type: "node", bundle: "article", field_types: BASE_FIELDS) {
+          label
+          fields {
+            id
+            label
+            description
+            type
+            required
+            multiple
+            maxNumItems
+            status
+            defaultValue
+            isReference
+            isHidden
+            weight
+            settings
+          }
+        }
+      }
+GQL;
+
+    $expectedResult = $this->fullDefinitionResult;
+    // Remove the one configured field that should not be in the list of base
+    // fields.
+    unset($expectedResult['entityDefinition']['fields'][18]);
```

Comment:
maybe we could split the list of expected fields result into two variable - base fields result and configurable fields result. then use `array_merge` if we need to compare both of them:
```
protected function getEntityDefinitionResult(string $type, array $result) {
  return [
    'entityDefinition' =>
      [
        'label' => $type,
        'fields' => $result,
      ]
  ];
}
$fullDefinitionResult = $this->getFullDefinitionResult('Content', array_merge($this->baseFieldResult, $this->configFieldResult));
```

but fine for me this way as well

Reply:
There is just one configured field, so I would leave it as is.